server/src/connectionHandlers/onReferences.ts

```diff
@@ -0,0 +1,60 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) 2023 Savoir-faire Linux. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import type * as LSP from 'vscode-languageserver/node'
+import { logger } from '../lib/src/utils/OutputLogger'
+import { analyzer } from '../tree-sitter/analyzer'
+
+export function onReferencesHandler (referenceParams: LSP.ReferenceParams): LSP.Location[] | null {
```

Comment:
/home/deribaucourt/Workspace/yocto-vscode/vscode/src/vs/editor/contrib/gotoSymbol/browser/goToCommands.ts
```

			let altAction: SymbolNavigationAction | null | undefined;
			if (references.referenceAt(model.uri, position)) {
				const altActionId = this._getAlternativeCommand(editor);
				if (!SymbolNavigationAction._activeAlternativeCommands.has(altActionId) && SymbolNavigationAction._allSymbolNavigationCommands.has(altActionId)) {
					altAction = SymbolNavigationAction._allSymbolNavigationCommands.get(altActionId)!;
				}
			}

			const referenceCount = references.references.length;

			if (referenceCount === 0) {
				// no result -> show message
				if (!this.configuration.muteMessage) {
					const info = model.getWordAtPosition(position);
					MessageController.get(editor)?.showMessage(this._getNoResultFoundMessage(info), position);
				}
			} else if (referenceCount === 1 && altAction) {
				// already at the only result, run alternative
				SymbolNavigationAction._activeAlternativeCommands.add(this.desc.id);
				instaService.invokeFunction((accessor) => altAction!.runEditorCommand(accessor, editor, arg, range).finally(() => {
					SymbolNavigationAction._activeAlternativeCommands.delete(this.desc.id);
				}));

			} else {
				// normal results handling
				return this._onResult(editorService, symbolNavService, editor, references, range);
			}
```

Reply:
It shouldn't be using ReferencesAction in this case, instead DefinitionAction which provides it's _getLocationModel that provides references itself

Reply:
Turns out it's the behavior of a weird setting. CHanging it to evealDefinition makes our definition work without this patch.
![image](https://github.com/yoctoproject/vscode-bitbake/assets/19970889/ba7c24ca-4c5f-43db-8d1c-52014982fbf8)

Reply:
Since we don't plan to fully implement references, I would suggest to not write so many new lines of code as a workaround for this weird setting. I pushed a new commit which simply uses the definitions provider for references. It is correct though not exhaustive and fixes this issue without a whole new class. What do you think?

Reply:
I was planning to use the `onDefinitionHandler` on the `onReference` event, but I wasn't sure if it was right for bitbake. Since you did so, we should see how it goes.

Reply:
The VSCode philosophy is:
 - definition: where the variable is defined (first assignment)
 - reference: where the variable is used or modified
In bitbake, we can't distinguish the two because assignments and modifications are usually intertwined. It's the case in most languages. So for bitbake we can consider definitions to be references. If we want to go further, we could also look for more references like "${MYVAR}" references. Though it is not planned yet.